hardware/raspberrypi/bootmodes/gpio.md

```diff
@@ -0,0 +1,27 @@
+# GPIO boot mode
+
+The Raspberry Pi can be configured to allow the boot mode to be selected at run time using hardware attached to the GPIO connector: this is GPIO boot mode. This is done by setting a bit in the OTP memory of the SoC. Once the bit is set, it permanently allocates 5 GPIOs to allow this selected to be made. Once the OTP bit is set it cannot be undone, so you should think carefully about enabling this, since those 5 GPIO lines will then be permanently unavailable for any other use.
+
+To enable GPIO boot mode, add the following line to the config.txt file:
+
+```
+program_gpio_bootmode=n
+```
+
+And reboot the Pi once to program the OTP with this setting. Where n is the bank of GPIOs which you wish to use. Bank 1 is GPIOs 22-26, bank 2 is GPIOs 39-43. Unless you have a compute module, you must use bank 1: the GPIOs in bank 2 are only available on the compute module.
+
+Once GPIO boot mode is enabled, the Raspberry Pi will no longer boot. You must pull up at least one boot mode GPIO pin in order for the Pi to boot.
```

Comment:
Probably also worth explaining the priority-order, if multiple bootmodes are selected?

Reply:
@ghollingworth - is the boot order in order of GPIO number?

Reply:
Yes it is...